pkg/controller/scyllaclustermigration/migrate_scyllacluster.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2024 ScyllaDB.
+
+package scyllaclustermigration
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"github.com/scylladb/scylla-operator/pkg/controller/scylladbdatacenter"
+	"github.com/scylladb/scylla-operator/pkg/controllerhelpers"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/resourceapply"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (scmc *Controller) syncScyllaDBDatacenter(ctx context.Context, sc *scyllav1.ScyllaCluster) ([]metav1.Condition, error) {
+	var progressingConditions []metav1.Condition
+
+	sdc, upgradeContext, err := migrateV1ScyllaClusterToV1Alpha1ScyllaDBDatacenter(sc)
+	if err != nil {
+		return progressingConditions, fmt.Errorf("can't migrate scyllav1.ScyllaCluster to v1alpha1.ScyllaDBDatacenter: %w", err)
+	}
+
+	existingSDc, err := scmc.scyllaDBDatacenterLister.ScyllaDBDatacenters(sc.Namespace).Get(sc.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return progressingConditions, fmt.Errorf("can't get v1alpha1.ScyllaDBDatacenter %q: %w", naming.ManualRef(sc.Namespace, sc.Name), err)
+	}
```

Comment:
this should read the map of owned objects

Reply:
done